Zmo/committee sites

koivyr opened the pull request:
### Summary
- Created the committee_sites. 
- Create a page editing system. One can edit the committee sites as well as create new pages if they have the privilege. 
- A user can upload files to the site if needed.  
- Currently, anyone can edit pages, but it can easily be edited such that only people with admin permission can access those links. 

### Test Plan
- Create a new page in the editor. Check that the page exists, and check that it exists in the 'all page created' page. Check that the file can be deleted. 
- Upload a file and check that it exists in the ‘all uploads’ page. Check that the file can be deleted.

koivyr committed:
Added link to Arc site

koivyr committed:
Working on BoC site

koivyr committed:
added folder for committees; working on creating table of people in BOC

koivyr committed:
added path to current committee_sites

koivyr committed:
file moved

koivyr committed:
changed routes, and helpers for new html files

koivyr committed:
add new BoC files

koivyr committed:
Add links for BoC pages

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Add links, make text pretty

koivyr committed:
Add routes

koivyr committed:
New folder for testing editor pages

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

Conflicts:
	donut/__init__.py

koivyr committed:
Add test data, make lint

koivyr committed:
Add test, update past tests

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Roll back to old tests; bug fixes

koivyr committed:
Markdown now works

koivyr committed:
Update Java script

koivyr committed:
Make Lint

koivyr committed:
Add folder and code to prepare for saving files

koivyr committed:
Prepare for uploading files

koivyr committed:
Uploading new files kind of works

koivyr committed:
You can create new pages now!!!

next steps:
Fix redirecting
Fix cases of trying to write to a file twice
Fix the JS for title...
CSS the editor page

koivyr committed:
Make Lint

koivyr committed:
Clean up

koivyr committed:
Re-did uploading files

koivyr committed:
Bug fixes

koivyr committed:
More bug fixes

Commit:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Css'ed the editor page a little.

TODO:
More css
add permissions
debug uploading file systems
create a page that will show links to all uploaded files
Debug editing current exisiting pages

koivyr committed:
Trying to figure out why uploads are not serving up files

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

Conflicts:
	donut/__init__.py

Commit:
Added a page for uploading

koivyr committed:
Add redirecting after creating a page.

Still need to figure out why optional parameters are not being read

Commit:
Add a page that will show all the existing created pages

davistardif committed:
Added error message for unsupported file extension

davistardif committed:
Partial fix for uploaded list

davistardif committed:
Fixed conflicts

davistardif committed:
Fixed upload / download

davistardif committed:
Begin fixing editor

Commit:
Fix bug where editor can't find the created webpages

davistardif committed:
Fixes to editor markdown buttons

davistardif committed:
Fixed dropdown hover

koivyr committed:
Begin to check for permissions for editing pages

koivyr committed:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites

davistardif committed:
Fixed uploading and viewing pages

davistardif committed:
fixed merge conflict

koivyr committed:
Users who have permission can edit existing pages

koivyr committed:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites

Conflicts:
	donut/__init__.py
	donut/modules/uploads/helpers.py
	donut/modules/uploads/routes.py

koivyr committed:
Start Cssing

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

Commit:
Commenting and cleaning up

Commit:
Users can edit pages if they are admin; more clean up

davistardif committed:
Fix broken link

davistardif committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites
Need to get Sean's changes to position / position holders

davistardif committed:
Should fix BoC members, pending actual data in the database

davistardif committed:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites
pull before push

Commit:
Refractor more; add button for lists

Commit:
Add testing

koivyr committed:
Fixed bugs arised from adding tests and data

Commit:
Add test for uploads

koivyr committed:
Fixed testing

davistardif committed:
Fixed get_BoC_member

davistardif committed:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites
pull before push

Commit:
Add a button to delete uploaded files

Commit:
Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Fixed delete

Commit:
Add delete page button

koivyr committed:
Fix bug with boc group links and get boc/get crc

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

Commit:
Add more tests

koivyr committed:
Fix tests

koivyr committed:
Make lint

koivyr committed:
Make lint

Commit:
Refractor and code clean up

koivyr committed:
Make lint

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Fix tests + again cleared code

koivyr committed:
Fix issues raised on github

calebsander commented on donut/modules/uploads/uploaded_files/pages/davis_page.md:
```diff
@@ -0,0 +1,3 @@
+test
+
+test
```
These test files should be removed. I think we should also add a `.gitignore` for the user-created `md` and `html` files.

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+    """
+    # get the group id of BoC
+    group_id = groups.get_group_list_data(['group_id'], {'group_name': name})
+    if group_id != []:
```
Put the error case first and return early:
````python
if not group_id: return []

# Here, group_id is non-empty
group_id = group_id[0]['group_id']
# ...
````

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+        INNER JOIN position_holders ph ON ph.pos_id=p.pos_id OR
+        pr.pos_id_from=ph.pos_id
+        WHERE group_id = %s) sub INNER JOIN positions p
+        ON p.pos_id = sub.pos_id'''
```
This should be integrated into the groups helpers module, since it duplicates a lot of the SQL already there

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+
+            (name, email) = core.get_name_and_email(members[i]['user_id'])
+            pos_name = result[i]['pos_name']
+            if pos_name == 'Chair' or 'ecretary' in pos_name:
```
This is kinda gross. Maybe make a named function that checks whether a position is a chair or sec?

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+        high_pos = []
+        pos_house = []
+        for i in range(len(result)):
+
```
Remove empty line

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+        members = core.get_member_data(members)
+        high_pos = []
+        pos_house = []
+        for i in range(len(result)):
```
`for member, position in zip(members, result)`

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+        ON p.pos_id = sub.pos_id'''
+        with flask.g.pymysql_db.cursor() as cursor:
+            cursor.execute(query, [group_id])
+            result = cursor.fetchall()
```
`result` is not a very descriptive variable name

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+            (name, email) = core.get_name_and_email(members[i]['user_id'])
+            pos_name = result[i]['pos_name']
+            if pos_name == 'Chair' or 'ecretary' in pos_name:
+                high_pos.append((name, pos_name, email))
```
Would be cleaner if these were `dict`s: `dict(name=name, pos_name=pos_name, email=email)`. Also, you use the same value in both the `if` and `else` blocks, so you should only compute it in one place.

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+        high_pos.sort(key=lambda tup: tup[1])
+        pos_house.sort(key=lambda tup: tup[1])
+
+        fin_result = high_pos + pos_house
```
Again, not a very descriptive variable name. Maybe `all_members`?

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+from donut.modules.core import helpers as core
+
+
+def get_member(name):
```
`def get_members(group_name)`

calebsander commented on donut/modules/committee_sites/helpers.py:
```diff
@@ -0,0 +1,47 @@
+
+def get_member(name):
+    """
+    Queries the database and returns list of people on BoC
```
Update comment; this is no longer `BoC`-specific; should also specify return format

calebsander commented on donut/modules/committee_sites/routes.py:
```diff
@@ -0,0 +1,74 @@
+    return flask.render_template(
+        'CRC.html',
+        CRC=helpers.get_member('CRC'),
+        permission=check_permission(Permissions.ADMIN))
```
Is there a way to make these dynamic rather than having to manually specify every page in the system? Something like a route of `/pages/<page>` which would then be looked up in the file system?

calebsander commented on donut/modules/committee_sites/static/script.js:
```diff
@@ -0,0 +1,10 @@
+
+
```
Remove blank lines

calebsander commented on donut/modules/committee_sites/templates/BoC.html:
```diff
@@ -0,0 +1,206 @@
+{% extends "layout.html" %}
```
If these pages are dynamically generated, they shouldn't be committed. Commit the Markdown used to make them instead.

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+import glob
+from flask import current_app, redirect, url_for
+
+underCommittee = [
```
Use a `set` for faster lookup

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+                           current_app.config["UPLOAD_WEBPAGES"],
+                           newfilename + '.md')
+    os.rename(oldpath, newpath)
+    return
```
`return` statement unnecessary

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+        path = os.path.join(flask.current_app.root_path,
+                            flask.current_app.config['UPLOAD_WEBPAGES'])
+        curFile = read_file(os.path.join(path, name + '.md'))
+        return curFile
```
Can put `return curFile` after the `if`-`else` statement

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+
+
+@blueprint.route('/editor', methods=['GET', 'POST'])
+def editor(input_text='Hello World!!!', title="TITLE"):
```
Remove default params

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    '''
+    Actually saves the data from the forms as markdown
+    '''
+    title = flask.request.form['title']
```
Appears unimplemented

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+        title = flask.request.args.get('title')
+
+    return flask.render_template(
+        'editor_page.html', input_text=input_text, title=title)
```
`title` will not be defined if `input` is `None` and a Flask error will be thrown

koivyr replied:
title would be "TITLE" still

calebsander replied:
That doesn't seem like a good default to have -- it should be an error if no title is specified

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+                        current_app.config["UPLOAD_WEBPAGES"])
+    links = glob.glob(root + '/*')
+    results = []
+    for i in range(len(links)):
```
`for file in links`. You don't actually use `i` anywhere.

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+    if title in underCommittee:
+        root = flask.current_app.config["COMMITTEE_UPLOAD_FOLDER"]
+        path = os.path.join(root, title + '.html')
+        f = open(path, "w+")
```
Use a `with` block

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+    path = os.path.join(root, title + ".md")
+
+    # Writing to the new html file
+    f = open(path, "w+")
```
Use a `with` block

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+    f.write(markdown)
+    f.close()
+
+    f = open(
```
Use a `with` block

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,115 @@
+        "w+")
+    f.write(title)
+    f.close()
+    return 0
```
Is this `return` value useful for anything?

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    Returns the editor page where users can create and edit
+    existing pages
+    '''
+    input = flask.request.args.get('input_text')
```
`input` is a built-in function; can you name the variable something else?

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    elif title == 'BoC_Witness':
+        return flask.jsonify({'url': url_for('committee_sites.winesses')})
+    elif title == 'CRC':
+        return flask.jsonify({'url': url_for('committee_sites.CRC')})
```
Ideally these would not be hard-coded

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    if (helpers.write_markdown(markdown, title) == 0):
+        return flask.jsonify({'url': url_for('uploads.display', url=title)})
+    else:
+        return flask.jsonify({'url': url_for('uploads.display', url=title)})
```
Your `if` and `else` blocks do the same thing

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    title = flask.request.form['title']
+
+
+@blueprint.route('/_save', methods=['POST'])
```
These are very general URLs. Can you prefix them with `/pages` or something?

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,76 @@
+    '''
+    filename = flask.request.form.get('filename')
+    if filename != None:
+        helper.remove_link(filename)
```
Remove all the code related to `filename` here; there is no `filename` specified in the URL

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,28 @@
+        {% for link, name in links %}
+        <tr>
+		<td><a href= "{{ link }}">{{name}}</a></td>
+    <td><a href= "{{ url_for('editor.created_list', filename=name)}}">Delete</td>
```
Indentation is a little messed up here

calebsander commented on donut/modules/editor/templates/redirecting.html:
```diff
@@ -0,0 +1,4 @@
+{% extends "layout.html" %} 
+{% block page %} 
+<a href="{{input_text}}">clickme!</a> 
+{% endblock %}
```
Is this file needed?

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,83 @@
+
+      <textarea id="text_title" rows="1" cols="5" >{{title}}</textarea>
+      <br/>
+      <button id="chang_title" onClick="change_title()">Change Title</button>
```
Should be `change_title`?

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,28 @@
+    <div class="jumbotron">
+      <h1 id = "title" class="text-center">List of User Created Pages</h1>
+      <div id = "content" style = "font-size:21px;font-weight:200;
+        margin-bottom:15px;">
```
Should try not to break HTML strings across lines

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+            type: 'POST',
+            data:{markdown:text, title:title},
+	    success: function(data) {
+            window.location.href = data['url']
```
`data.url`

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+    // Checking for valid titles; should not have
+    // anything other than numbers, characters, underscore,
+    // period, front slash, hyphen, and spaces.
+    var valid = /^[0-9a-zA-Z.\/_\- ]*$/.test(title);
```
Title should be validated on serverside if it matters

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+  while(ta.indexOf(number+".") > -1)
+  {
+    number = number + 1;
+  }
```
Why can't you just always insert `1. `? This would allow multiple `ol`s in the same file.

koivyr replied:
At some point, I wanted to be able to "guess" which order list the user is in. The markdown does allow for nested ordered lists, but it requires indentations. I wasn't sure how to implement that particular part.

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+
+function insert_olist(){
+  var number = 1;
+  var ta = document.getElementById("source").value.toString();
```
Should already be a string, no?

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+// All markdown related functions
+function insert_heading(size){
+  var string = "";
+  for(var i = 0; i< size; i++)
```
`i < size`

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+function selectedString(string1, string2)
+{
+  var ta = document.getElementById('source');
+  if ((ta.value.substring(ta.selectionStart, ta.selectionEnd)) != "")
```
Don't need the inner parens

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+  }
+  else {
+    insert(string2, true);
+  }
```
I would make the condition `==` and swap the `if` and `else` blocks to make it more readable

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+         url: $SCRIPT_ROOT+'/_change_title',
+         type: 'POST',
+         data:{title:title},
+  success: function(data) {
```
Indentation

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+         type: 'POST',
+         data:{title:title},
+  success: function(data) {
+        document.getElementById("chang_title").innerHTML = "Title Change Sucessfully";
```
Shouldn't this go away in case you want to change the title again?

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+      converter = new showdown.Converter({strikethrough: true}),
+      html = converter.makeHtml(text);
+
+    target.innerHTML = html;
```
Indendation

calebsander commented on donut/modules/editor/static/pages.txt:
```diff
@@ -0,0 +1 @@
+ANOTHER_TITLE
```
What is this used for?

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,138 @@
+}
+
+
+function insert(string, bool){
```
`bool` is not a descriptive name

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,28 @@
+{% extends "layout.html" %}
+{% block page %}
+<script src="https://cdn.rawgit.com/showdownjs/showdown/1.0.1/dist/showdown.min.js"></script>
+<script src={{ url_for('uploads.static', filename='script.js', _external = True)}} type="text/javascript"></script>
```
Add quotes around url string

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,83 @@
+              <li><a class="btn btn-outline-secondary btn-sm" type="button" onClick="insert_heading(3)" style="font-size:16px">Heading</a></li>
+              <li><a class="btn btn-outline-secondary btn-sm" type="button" onClick="insert_heading(4)" style="font-size:14px">Heading</a></li>
+              <li><a class="btn btn-outline-secondary btn-sm" type="button" onClick="insert_heading(5)" style="font-size:12px">Heading</a></li>
+              <li><a class="btn btn-outline-secondary btn-sm" type="button" onClick="insert_heading(6)" style="font-size:10px">Heading</a></li>
```
Could make these with a `for` loop in Jinja

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,60 @@
+    content = ''
+    with open(path, 'r') as f:
+        content += f.read()
+    return content
```
You only get the content once; could just write `return f.read()`

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,60 @@
+            dele[counter] = i
+            counter += 1
+    links.remove(dele[0])
+    links.remove(dele[1])
```
What is this `dele` and `counter` stuff doing?

koivyr replied:
Stupid way of getting rid of two directories. I fixed it

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,60 @@
+import glob
+
+
+def readPage(url):
```
Should be snake_case for consistency

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,71 @@
+        if file.filename == '':
+            flask.flash('No selected file')
+            return flask.redirect(request.url)
+        if file and helpers.allowed_file(file.filename):
```
Won't `file` always be truthy, since you just tried to access its `filename` property?

calebsander commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,9 @@
+      html = converter.makeHtml(input);
+  console.log(html);
+  console.log(input);
+	  console.log(target.innerHTML);
```
Remove `console.log`s

calebsander commented on donut/modules/uploads/templates/Hello World!!!.html:
```diff
@@ -0,0 +1 @@
+<p>Hello World!!!</p>
```
This file can probable be removed

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,20 @@
+    </div>
+    {% if permission %}
+    <a href="{{ url_for('editor.editor', input_text=title, title=title) }}">edit this page</a>
+    {% else %}
```
Remove empty `else` block

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,20 @@
+{% extends "layout.html" %}
+{% block page %}
+<script src="https://cdn.rawgit.com/showdownjs/showdown/1.0.1/dist/showdown.min.js"></script>
+<script src={{ url_for('uploads.static', filename='script.js', _external = True)}} type="text/javascript"></script>
```
Put quotes around `src` URL

calebsander commented on donut/modules/uploads/templates/template_page:
```diff
@@ -0,0 +1,5 @@
+{% block page %}
+<div class="container">
+  <div class="col-md-24">
+    <div class="jumbotron">
```
This file looks like it should be deleted

calebsander commented on tests/modules/committee_sites/test_committee_sites.py:
```diff
@@ -0,0 +1,41 @@
+from donut import app
+import donut.modules.core.helpers as core_helpers
+from donut.modules.committee_sites import helpers
+from donut.modules.committee_sites import routes
```
You can combine these two imports: `from donut.modules.committee_sites import helpers, routes`

calebsander commented on tests/modules/editor/test_editor.py:
```diff
@@ -0,0 +1,61 @@
+    helpers.remove_link("TEST_TITLE")
+    helpers.remove_link("ANOTHER_TITLE")
+    links = helpers.get_links()
+    titles = []
```
`titles = [title for _, title in links]`

calebsander replied:
This was addressed in 1 place but not the other 2

calebsander commented on tests/modules/uploads/test_uploads.py:
```diff
@@ -0,0 +1,59 @@
+                        "SOME_TITLE" + ".jpg")
+
+    f = open(path, "w+")
+    f.write("")
```
Don't think `write()` is necessary; file will already be cleared

koivyr committed:
Fixed a lot styling + concerns listed on the PR

koivyr committed:
Merge branch 'master' into zmo/committee_sites

koivyr committed:
Make Lint

koivyr committed:
Feedback from Caleb

koivyr committed:
make lint

koivyr committed:
Cleaned CRC webpage

koivyr committed:
Hopefully fixed all of the things on the PR..

koivyr commented:
@calebsander ? ty

calebsander commented on donut/__init__.py:
```diff
@@ -25,7 +27,10 @@
+app.register_blueprint(editor.blueprint)
 app.register_blueprint(rooms.blueprint)
+app.register_blueprint(uploads.blueprint)
+app.register_blueprint(directory_search.blueprint)
```
This is already registered

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,87 @@
+    '''
+    Reads in a file
+    '''
+    curFile = ''
```
Unused variable

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,87 @@
+    path = os.path.join(root, title + ".md")
+
+    # Writing to the new html file
+    with open(path, 'w+') as f:
```
`+` isn't needed if you're just writing

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,126 @@
+// Changes the title of a file
+function change_title(){
+  var title = document.getElementById('title').value;
+  var valid = /^[0-9a-zA-Z.\/_\- ]*$/.test(title);
```
If `valid` is `false`, the `POST` should not be sent

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,69 @@
+    page = helpers.read_page(url.replace(' ', '_'))
+    return flask.render_template(
+        'page.html', page=page, title=url.replace('_', ' '), permissions=True)
+    # permission=check_permission(Permissions.ADMIN))
```
Either uncomment this or remove it from the code

koivyr replied:
So a lot of these are left on so that you can test these features since the permission system itself isn't flushed out. Otherwise only Permission.ADMIN people can do edit/create/delete things

calebsander replied:
Hmm, can we at least require the user to be logged in?

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,69 @@
+
+
+@blueprint.route('/uploads', methods=['GET', 'POST'])
+def uploads():
```
Would be cleaner to add separate routes for `GET /uploads` and `POST /uploads`

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,69 @@
+        if 'file' not in flask.request.files:
+            flask.flash('No file part')
+            return flask.render_template('uploads.html')
+        file = flask.request.files['file']
```
Can we check that the user is logged in and the file is of a reasonable size (e.g. < 10 MB)? Otherwise it would be very easy to fill up the EC2 instance's hard drive.

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,69 @@
+            filename = secure_filename(file.filename)
+            uploads = os.path.join(flask.current_app.root_path,
+                                   flask.current_app.config['UPLOAD_FOLDER'])
+            file.save(os.path.join(uploads, filename))
```
How do we prevent overwriting other files with the same name?

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,19 @@
+      <h1 id = "title" class="text-center">{{title}}</h1>
+      <div id = "content">
+
+      <script>convert(`{{page}}`);</script>
```
JS strings should use quotes; the backtick is only supported in modern versions of JS

koivyr replied:
So I have a pretty ugly way around this, but honestly it would be preferable to be able to have multi line comments

calebsander replied:
Couldn't you just have the JS make an AJAX call to a Flask endpoint that sends it the contents?

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,19 @@
+
+      <script>convert(`{{page}}`);</script>
+    </div>
+    {% if permissions%}
```
Permissions should be enforced on server too

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,60 @@
+
+
+@blueprint.route('/editor', methods=['GET', 'POST'])
+def editor(input_text='Hello World!!', title="TITLE"):
```
Remove these variable declarations

calebsander reviewed (changes requested):
I added review comments. There are still some left from a previous review. Generally, I have a lot of security concerns about these changes; there seems to be no restriction to users who have permissions to edit these pages and upload files. Could we talk about this at some meeting?

koivyr committed:
WIP: Fixing more concerns in PR

Mostly changed things regard security and accessing
certain pages. Unfortunately this means that some of the route
testing had to be changed to 403, permission denied.

I still need to fix the style and shuffle things from routes to
helper

koivyr committed:
WIP:I missed a file

koivyr committed:
fix styling

koivyr committed:
Make lint

koivyr committed:
Fix bug where anyone can delete an uploaded file

koivyr committed:
Fix more permission/security concerns

Fix bugs related to deleting pages

koivyr committed:
bug set permission to true

koivyr committed:
Strict check on deleting filenames

rlin0 commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+      data:{markdown:text, title:title},
+      success: function(data) {
+        if (data.error === "")
+	{
```
indentation

rlin0 commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+        }
+	else if (data.error === "Duplicate title") {
+	  var res = confirm("You are overriding an existing file!");
+          if (res){
```
indentation off

rlin0 commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,84 @@
+            <span class="caret"></span></button>
+            <ul class="dropdown-menu">
+
+       		 {% for i in range(1,7) %}
```
indentation

rlin0 commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,28 @@
+	  <td><a href= "{{ link }}">{{name}}</a></td>
+	  {% if permissions%}
+   	    <td><a href= "{{ url_for('editor.created_list', filename=name)}}">Delete</td>
+       	  {% endif %}
```
indent

rlin0 commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,84 @@
+        <div class="btn-group" role="group">
+          <div class="dropdown">
+            <butt
+            on class="btn btn-outline-primary btn-sm" type="button">Lists
```
button

rlin0 commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,72 @@
+            url: $SCRIPT_ROOT+'/_upload_file',
+            type: 'POST',
+  	        data:form_data,
+  		      processData: false,
```
indent

rlin0 commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,72 @@
+              processData: false,
+              contentType: false,
+  	          success: function(data) {
+  		          window.location.href = data.url;
```
indent

joshc commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,83 @@
+    '''
+    markdown = flask.request.form['markdown']
+    title = flask.request.form['title']
+    title_res = re.match("^[0-9a-zA-Z.\/_\- ]*$", title)
```
You check the title multiple times, also in script.js:change_title(). What is the proper place to check (try to avoid redundant checks)?

koivyr replied:
I'll change it to only check in the server then.

joshc replied:
also comment what this matches for future-proofing.

joshc commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+
+// Changes the title of a file
+function change_title(oldTitle){
+  var title = document.getElementById('title').value;
```
this should get 'text_title' right? otherwise, the validity check below will always fail with undefined.

rlin0 commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+}
+
+function insert_image(){
+  insert("![Alt text](url/to/image", false);
```
) after image

rlin0 commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,75 @@
+import glob
+
+ALLOWED_EXTENSIONS = set(
+    ['docx', 'doc', 'txt', 'pdf', 'png', 'jpg', 'jpeg', 'gif'])
```
Why is this a set?

joshc replied:
fast(er) lookup for sets, can't have duplicates :D

rlin0 commented on donut/templates/includes/header.html:
```diff
@@ -39,10 +39,10 @@
-                            <li><a href="#">CRC</a></li>
-                            <li><a href="#">IHC</a></li>
+                            <li><a href="https://sites.google.com/site/arccaltech/">ARC</a></li>
+			    <li><a href="{{ url_for('uploads.display', url='BoC') }}">BoC</a></li>
```
indent

rlin0 commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+  if(selected)
+  {
+    var txt2 = ta.value.substring(0, ta.selectionStart)
+  + string + ta.value.substring(ta.selectionStart, ta.selectionEnd) + string
```
indent

joshc commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,84 @@
+      <hr/>
+      <button id="runBtn" onClick="run()">preview</button>
+      <button id="save" onClick="save()">Save Changes</button>
+      <div id="preview">
```
Add a header for preview :)

rlin0 commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,156 @@
+  var ta = document.getElementById('source');
+  if(selected)
+  {
+    var txt2 = ta.value.substring(0, ta.selectionStart)
```
You can directly do ta.value = ... in the if else statements, don't need var txt2

rlin0 commented on donut/modules/editor/static/style.css:
```diff
@@ -0,0 +1,19 @@
+
+}
+.btn{
+    margin: 5px;
```
indent

rlin0 commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,28 @@
+        {% for link, name in links %}
+        <tr>
+	  <td><a href= "{{ link }}">{{name}}</a></td>
+	  {% if permissions%}
```
nitpick : space after permissions

joshc commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,105 @@
+    '''
+    Displays the webpages that have been created by users.
+    '''
+    page = helpers.read_page(url.replace(' ', '_'))
```
2 comments.

One, it seems like [http://127.0.0.1:9000/lib/BoC_Bylaws](http://127.0.0.1:9000/lib/BoC_Bylaws) and [http://127.0.0.1:9000/lib/BoC Bylaws](http://127.0.0.1:9000/lib/BoC Bylaws) and [http://127.0.0.1:9000/lib/BoC%20Bylaws](http://127.0.0.1:9000/lib/BoC%20Bylaws) all go to the same page because of this replace. Hopefully this is the intended behavior.

Two, you should better handle it if a page does not exist (i.e. has not been created). Right now, loading a page like [http://127.0.0.1:9000/lib/BoCCoB](http://127.0.0.1:9000/lib/BoCCoB) gives a FileNotFound error on the server rather than displaying a Page not found error on the website itself (which for some reason is handled correctly when trying to access uploaded files that were not created).

joshc commented on donut/__init__.py:
```diff
@@ -55,7 +59,10 @@ def init(environment_name):
     app.config["DB_PASSWORD"] = environment.db_password
     app.config["DB_NAME"] = environment.db_name
+    app.config["UPLOAD_WEBPAGES"] = 'modules/uploads/uploaded_files/pages'
+    app.config["UPLOAD_FOLDER"] = 'modules/uploads/uploaded_files'
```
Consider adding these directories to the .gitignore, so that uploaded files are not committed to the git repo. See [this](https://stackoverflow.com/questions/4250063/how-to-gitignore-all-files-folder-in-a-folder-but-not-the-folder-itself) to maintain uploaded_files/pages directory (basically local .gitignore).

koivyr committed:
Still have one last bug to fix

Need to fix bug where overriding upload files
do not actually override
You can use control-Z/Y on the editor now
There is a new button for the editor page if
you are logged in and an admin
PR feedback

koivyr committed:
Created documents page

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Still need to change the styles

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -1,26 +1,28 @@
+<head>
 <script src="https://cdn.rawgit.com/showdownjs/showdown/1.0.1/dist/showdown.min.js"></script>
 <script src="{{ url_for('uploads.static', filename='script.js', _external = True)}}" type="text/javascript"></script>
+</head>
```
`<head>` tag should not go inside `<body>`, which is where `{% block page %}` is inserted. Put scripts in `{% block scripts %}` instead.

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -1,27 +1,42 @@
+  <script type="text/javascript">
+  $SCRIPT_ROOT = {{ request.script_root|tojson|safe }};
+  </script>
+</head>
```
As above, `<link rel="stylesheet">` should go in `{% block styles %}` and `<script>` should go in `{% block scripts %}`

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -1,27 +1,42 @@
       <div id='sucesss'>
       </div>
-
+      <p id="old_title" hidden>{{title}}</p>
```
What is this needed for if it's `hidden`?

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -11,8 +12,11 @@ def read_page(url):
+        with open(path, 'r') as f:
+            return f.read()
+    else:
+        return -1
```
`-1` doesn't make a lot of sense. Maybe return `None` or `''` instead?

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -31,9 +35,10 @@ def remove_link(filename):
         if filename == name:
-            os.remove(link)
+            print(os.remove(link))
+            time.sleep(1)
```
Remove these 2 lines

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -13,11 +14,14 @@ def display(url):
+    if page == -1:
+        return flask.abort(404)
+    else:
+        return flask.render_template(
```
This doesn't need to be nested inside an `else` block

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -52,8 +56,10 @@ def upload_file():
     if 'username' in flask.session and check_permission(Permissions.ADMIN):
+        helpers.remove_link(filename)
+        file.save(os.path.join(uploads, filename))
+        time.sleep(5)
```
Remove `time.sleep()`

calebsander commented on donut/modules/uploads/uploaded_files/pages/2018-2019_Minutes.md:
```diff
@@ -0,0 +1,9 @@
+
+      
+
+      
```
Let's not put uploads in the git repository

koivyr replied:
Hmm I was thinking of keeping a few around for examples. Also, the documents tab has to be populated with something.

calebsander replied:
Sure, examples are fine, but I don't want the git repo to be the place where actual uploads go. They should just live on the server.

calebsander replied:
I don't think this was addressed. These files will have to be temporarily removed from the `.gitignore` so their deletion can be committed.

koivyr replied:
Remove all the .md files? This would include the documents pages and all boc/crc related pages...

calebsander replied:
Hmm, if you think the page is *very* unlikely to be edited, I think it's okay to include it. But if the page might change, we probably don't want to have to keep the repo up to date with the copy on the server.

calebsander commented on donut/modules/uploads/templates/uploads.html:
```diff
@@ -1,12 +1,12 @@
+  <script src={{ url_for('uploads.static', filename='script.js', _external = True)}} type="text/javascript"></script>
+  <script type="text/javascript">
+    $SCRIPT_ROOT = {{ request.script_root|tojson|safe }};
+  </script>
```
Again, `<head>` shouldn't be inside `<body>`

calebsander commented on donut/templates/includes/header.html:
```diff
@@ -40,10 +40,13 @@
                             <li><a href="{{ url_for('uploads.display', url='CRC') }}">CRC</a></li>
                             <li><a href="http://ihc.caltech.edu/">IHC</a></li>
                             <li><a href="#">ESC</a></li>
+                            {% if 'permissions' in session and 'username' in session and 1 == session['permissions'] %}
```
I don't think we're using `session['permissions']` to store any data

koivyr replied:
Hmmm, do you think it'll be a good idea to store something within session['x_permissions'] so that in the layout.html, you can check if a certain tab has to be generated? I'm not sure if there are other ways around this?

calebsander replied:
Can't you just pass a variable when you call `render_template()` that says whether or not to show the tab?

koivyr replied:
The header is a part of layout.html, which is being extended in every page.

calebsander replied:
Hmm, maybe something like that would make sense, but I definitely don't think we want to be storing the permissions in the session storage. If we set `session['permissions']`, can we add `del session['permissions']` to `teardown_request()` in `donut/__init__.py`?

calebsander commented on donut/default_permissions.py:
```diff
@@ -11,3 +11,4 @@
 class Permissions(enum.IntEnum):
     #Site admins -- always have permission to everything -- Use with caution
     ADMIN = 1
+    EDIT = 2
```
Should go in a separate module-specific permissions enum

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -65,6 +65,8 @@ def remove_link(filename):
     for i in links:
         name = i.replace(path + '/', '').replace('.md', '').replace("_", " ")
+        print(name)
+        print(filename)
```
Remove these

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -77,13 +79,17 @@ def created_list():
+            'created_list.html',
+            links=links,
+            permissions=check_permission(flask.session['username'],
+                                         Permissions.EDIT))
```
````python
return flask.render_template(
    'created_list.html',
    links=links,
    permissions=('username' in flask.session and 
        check_permission(flask.session['username'], Permissions.EDIT)))
````

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -34,10 +34,13 @@ def remove_link(filename):
     for link in links:
         name = link.replace(path + '/', '')
+        print(name)
+        print(filename)
```
Remove `print` statements

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -17,11 +17,20 @@ def display(url):
+                'page.html',
+                page=page,
+                title=url.replace('_', ' '),
+                permissions=False)
```
````python
return flask.render_template(
    'page.html',
    page=page,
    title=url.replace('_', ' '),
    permissions=('username' in flask.session and
        check_permission(flask.session['username'], Permissions.EDIT)))
````

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -100,12 +112,18 @@ def uploaded_list():
-    if filename != None and check_permission(
-            Permissions.ADMIN) and 'username' in flask.session:
+    if filename != None and 'username' in flask.session and check_permission(
+            Permissions.EDIT):
```
Missing the username in a call to `check_permission()`

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -100,12 +112,18 @@ def uploaded_list():
+            'uploaded_list.html',
+            links=links,
+            permissions=check_permission(flask.session['username'],
+                                         Permissions.EDIT))
```
Again, these can be one call to `flask.render_template()`

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
Fixing permissions and feedback

calebsander commented on donut/modules/editor/edit_permission.py:
```diff
@@ -0,0 +1,5 @@
+import enum
+
+
+class Edit_permission(enum.IntEnum):
```
Nit: standard Python casing is `EditPermission`

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -1,9 +1,11 @@
-<script src="{{ url_for('uploads.static', filename='script.js', _external = True)}}" type="text/javascript"></script>
-</head>
+{% block scripts %}
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Including jQuery shouldn't be necessary since it's already included on every page

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -1,13 +1,19 @@
+    {{super()}}
+    <link rel="stylesheet" href={{ url_for('editor.static', filename='style.css')}}>
+  {% endblock %}
+  {% block scripts %}
```
Should include `{{ super() }}`

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -64,8 +69,7 @@ <h1 id = "title" class="text-center">Editing</h1>
-            <butt
-            on class="btn btn-outline-primary btn-sm" type="button">Lists
+            <button class="btn btn-outline-primary btn-sm" type="button">Lists
             <span class="caret"></span></button>
```
Indent pls; this `<button>` is hard to read

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -1,24 +1,30 @@
-      <a href="{{ url_for('editor.editor', input_text=title.replace(' ', '_'), title=title.replace(' ', '_'))}}">edit this page</a>
-    {% endif %}
+      <div id = "content" onload="load('{{title}}')">
+        <script> load('{{title}}');</script>
```
`load('{{title}}')` shouldn't be called in 2 different places

koivyr committed:
Feedback + a few more tests

koivyr committed:
Forgot to get rid of permission

koivyr committed:
Fix formatting

koivyr commented:
@calebsander @joshc?

rlin0 commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+    ) and check_permission(flask.session['username'], EditPermission.ABLE):
+        helpers.write_markdown(markdown, title)
+        return flask.jsonify({'url': url_for('uploads.display', url=title)})
+
```
Was this extra line intentional?

rlin0 commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,82 @@
+    if os.path.isfile(path):
+        with open(path, 'r') as f:
+            return f.read()
+    else:
```
don't need else

calebsander replied:
Ya, could do something like
```python
if not os.path.isfile(path):
    return None

with open(path) as f: # note that mode is 'r' by default
    return f.read()
```

rlin0 commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,68 @@
+  	    success: function(data) {
+              window.location.href = data.url;
+            },
+            error: function(data){
```
indentation is off

koivyr committed:
Start work on file lock system

calebsander commented on donut/__init__.py:
```diff
@@ -13,9 +13,11 @@
+from donut.modules import account, auth, marketplace, core, directory_search, editor, groups, rooms, uploads, voting
 
 app = flask.Flask(__name__)
+
```
Can you undo adding these 2 blank lines?

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,97 @@
+import flask
+import os
+import glob
+from flask import current_app, redirect, url_for
```
`redirect` and `url_for` are unused

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,97 @@
+        with open(path) as f:
+            return f.read()
+    else:
+        return ""
```
Might be cleaner to show that `not os.path.isfile(path)` is an exceptional case:
```python
if not os.path.isfile(path):
    return ''

with open(path) as f:
    return f.read()
```

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,97 @@
+    '''
+
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
```
For consistency, you should either use `flask.current_app` or `current_app` everywhere. I would suggest only importing one or the other.

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,97 @@
+        name = i.replace(path + '/', '').replace('.md', '')
+        if filename == name:
+            return "Duplicate title"
+    return ""
```
Is returning these special strings a good interface for the function? Perhaps returning a `bool` would make more sense

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+    '''
+    input_text = "Hello world"
+    title = "TITLE"
+    inputt = flask.request.args.get('input_text')
```
Passing markdown should probably go in the request body (i.e. a `POST` request); it could easily be too long to fit in a URL

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+        return flask.abort(403)
+
+
+@blueprint.route('/_change_title', methods=['POST'])
```
This is not a super RESTful URL. Should go under `/pages` or something

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+    title = flask.request.form['title']
+    # Allows all numbers and characters. Allows ".", "_", "-"
+    title_res = re.match("^[0-9a-zA-Z.\/_\- ]*$", title)
+    if title_res != None and len(title) <= 35 and check_login(
```
Doesn't seem like having a title that is too long should result in a 403

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+    return flask.jsonify({'error': helpers.check_duplicate(title)})
+
+
+@blueprint.route('/created_list')
```
Again, should be a sub-route of some other path

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,90 @@
+        'created_list.html',
+        links=links,
+        permissions=(check_login() and check_permission(
+            flask.session['username'], EditPermission.ABLE)))
```
Avoid calculating the permissions twice in this function. Would make sense to have some helper that encapsulates this condition

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+  var text = document.getElementById('source').value,
+      title = document.getElementById('text_title').value,
+      target = document.getElementById('preview'),
+      target_title = document.getElementById('preview_title'),
```
I would recommend using jQuery instead

koivyr replied:
I'm not terribly familiar with js; is there a reason?

calebsander replied:
It's generally a lot more concise, e.g. this section would become:
```js
 var text = $('#source').val(),
      title = $('#text_title').val(),
      target = $('#preview'),
      target_title = $('#preview_title'),
```

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+  var title = document.getElementById('text_title').value;
+  var input_text = document.getElementById('source').value;
+  $.ajax({
+    url: $SCRIPT_ROOT+'/_change_title',
```
Is `$SCRIPT_ROOT` really necessary? If you start your URL with `/`, it will be relative to the domain, so I think you just need `/_change_title`

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+function selectedString(string1, string2)
+{
+  var ta = document.getElementById('source');
+  if (ta.value.substring(ta.selectionStart, ta.selectionEnd) == "")
```
This condition is equivalent to `ta.selectionStart === ta.selectionEnd`, so you can avoid computing a substring

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+
+//###########
+
+function selectedString(string1, string2)
```
Function name and parameters could be more descriptive

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+}
+
+function insert_link(){
+  insert("[linkTitle](example.com)", false);
```
`Link title` would be more readable

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+      document.execCommand('insertText', false, string);
+    } else{
+      ta.value = ta.value.substring(0, ta.selectionStart)
+      + string + ta.value.substring(ta.selectionStart);
```
Use the `begin` and `end` values computed above

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+  if (title === '')
+  {
+    window.alert("Enter a title for your new page!");
+  }
```
Would be cleaner to just do an early return

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,166 @@
+              error: function(data){
+                window.alert("Please enter a valid title");
+              }
+            });
```
This case shares all this AJAX code with the first case, so they should be combined

calebsander commented on donut/modules/editor/static/style.css:
```diff
@@ -0,0 +1,18 @@
+  margin: 30px;
+  width: 80%;
+}
+.btn{
```
Nit: missing space between selector and `{`

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,31 @@
+  <div class="col-md-24">
+    <div class="jumbotron">
+      <h1 id = "title" class="text-center">List of User Created Pages</h1>
+      <div id = "content" style = "font-size:21px;font-weight:200; margin-bottom:15px;">
```
Put this in a stylesheet, or at least, a `<style>` tag

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,97 @@
+	  You can upload a file <a href="{{ url_for('uploads.uploads') }}">here</a>.
+	</p>
+	<p>
+	  You can use numbers, characters, underscores, and periods in the titles. Titles may not be longer than 35 characters.
```
35 characters seems very arbitrary. Why is there any need to limit title lengths?

koivyr replied:
I think someone at some point mentioned that people might, for fun, try to have a ridiculously long title... We can get rid of the limit if you feel that this isn't a concern.

calebsander replied:
Hmm, 35 characters is very short. Could we at least allow 100 or something?

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,97 @@
+	</p>
+	<p>
+	  You can use numbers, characters, underscores, and periods in the titles. Titles may not be longer than 35 characters.
+	  Titles with underscores are interchangeable with titles with spaces (IE, "This title" and "This_title"
```
`i.e.`

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,97 @@
+        <div class="btn-group" role="group">
+          <div class="dropdown">
+            <button class="btn btn-outline-primary btn-sm" type="button">Headings
+            <span class="caret"></span></button>
```
```html
<button ...>
  Headings
  <span class="caret"></span>
</button>
```

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,97 @@
+      </div>
+      <br/>
+
+      <textarea id="source" rows="10" cols="20" width="80%">{{input_text}}
```
`input_text` needs to be sanitized, e.g. if it contains `<` characters

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,97 @@
+
+      </textarea>
+      <hr/>
+      <button id="runBtn" onClick="run()">preview</button>
```
Capitalize "Preview". Also, `run()` could be named more descriptively

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,82 @@
+    Get rid of matching filenames
+    '''
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_FOLDER'])
```
Again, you are using `flask.current_app` and `current_app`. Pick one and stick with it

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,82 @@
+    for link in links:
+        name = link.replace(path + '/', '')
+        if filename == name:
+            os.remove(link)
```
Could use a `break` here to make this faster

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,82 @@
+    processed_links = []
+    for link in links:
+        filename = os.path.basename(link)
+        if filename != 'static' and filename != 'pages' and filename != '':
```
These special values shouldn't be hard-coded

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,115 @@
+    '''
+    url = flask.request.args.get('url')
+    page = helpers.read_page(url.replace(' ', '_'))
+    return flask.jsonify(result=page)
```
Use dictionary syntax for clarity: `flask.jsonify({'result': page})`. Actually, should probably just sent the result back as text so it doesn't have to parsed as JSON.

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,115 @@
+                flask.session['username'], UploadPermissions.ABLE)))
+
+
+@blueprint.route('/_send_page', methods=['GET'])
```
Again, this should be under some subpath, e.g. `/uploads`

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,115 @@
+        'uploaded_list.html',
+        links=links,
+        permissions=(check_login() and check_permission(
+            flask.session['username'], UploadPermissions.ABLE)))
```
Again, avoid computing `permissions` twice

calebsander commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,68 @@
+    convert(data.result)
+  });
+}
+$(function() {
```
This does not make sense. `$` doesn't take a function as an argument

koivyr replied:
It's shorthand for $(document).ready(function() { ... });

calebsander commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,68 @@
+                window.alert("Please enter a valid title");
+              }
+            });
+          }
```
Again, these two cases duplicate some code

calebsander commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,68 @@
+  });
+}
+$(function() {
+  $('#submit_btn').on('click', function(){
```
You should be consistent—either attach click handlers in HTML or JavaScript, don't do some of each. Personally, I prefer doing it in JS so the HTML is cleaner.

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,27 @@
+    <div class="jumbotron">
+      <h1 id = "title" class="text-center">{{title}}</h1>
+      <div id = "content">
+        <script> load('{{title}}');</script>
```
`<script>` tags shouldn't be embedded in HTML elements

calebsander commented on donut/modules/uploads/templates/page.html:
```diff
@@ -0,0 +1,27 @@
+  {{super()}}
+  <!-- Jquery is here because this page requires jqery and my js file to load-->
+  <!-- Putting this in a script block here screws up the bootstrap -->
+  <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
```
What do you mean by this? Your page is now going to include jQuery in 2 places since `{% block scripts %}` already contains the following:
```html
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js"></script>
<script src="/static/js/copyright.js"></script>
```
I think the issue you are having is likely because you have put these `<script>` tags in `<head>`, so they will execute before the other scripts are executed in `<body>`

calebsander commented on donut/modules/uploads/templates/uploaded_list.html:
```diff
@@ -0,0 +1,33 @@
+{% extends "layout.html" %}
+{% block scripts %}
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+  {{super()}}
```
We should not be loading 2 different versions of jQuery

calebsander commented on donut/modules/uploads/templates/uploaded_list.html:
```diff
@@ -0,0 +1,33 @@
+    <div class="jumbotron">
+      <h1 id = "title" class="text-center">List of Uploaded Pages</h1>
+      <div id = "content" style="font-size:21px;font-weight:200;
+        margin-bottom:15px;">
```
String should not span multiple lines. But this style in a `<style>` block or `css` file

calebsander commented on donut/modules/uploads/templates/uploaded_list.html:
```diff
@@ -0,0 +1,33 @@
+            <tr>
+              <td><a href= "{{ link }}">{{name}}</a></td>
+              {% if permissions%}
+                <td><a href= "{{ url_for('uploads.uploaded_list', filename=name)}}">Delete</td>
```
How does this URL cause the file to be deleted?

koivyr replied:
The function uploaded_list looks for a filename from flask.request.args.get. If there are something, it deletes it.

calebsander replied:
Hmm, that doesn't seem very intuitive. It might be clearer if you had a separate endpoint that did the deletion and then redirected the client to `uploads.uploaded_list`.

calebsander commented on donut/modules/uploads/templates/uploads.html:
```diff
@@ -0,0 +1,33 @@
+{% extends "layout.html" %}
+{% block scripts %}
+  {{super()}}
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Same issue with loading 2 copies of jQuery

calebsander commented on donut/modules/uploads/templates/uploads.html:
```diff
@@ -0,0 +1,33 @@
+	  If you are uploading a file with the same name, you may need to refresh the page before seeing
+	  the new file. 
+	</p>
+      </div>
```
Indentation of these elements is hard to parse

calebsander commented on donut/modules/uploads/templates/uploads.html:
```diff
@@ -0,0 +1,33 @@
+      </div>
+        <form id="upload-file" method="post" enctype="multipart/form-data">
+          <label for="file">Select a file</label>
+          <input name="file" type="file">
```
Tags without a close tag should look like `<input ... />` for readability

calebsander commented on donut/templates/includes/header.html:
```diff
@@ -35,14 +35,18 @@
                             <li><a href="{{ url_for('voting.list_surveys') }}">Surveys and Voting</a></li>
+                            {% if 'username' in session %}
+                              <li><a href="{{ url_for('editor.editor')}}">Create A New Page</a></li>
+                            {% endif %}
```
But not everyone can create a page, right? I think this link should be on the documents page instead

calebsander commented on tests/modules/uploads/test_uploads.py:
```diff
@@ -0,0 +1,105 @@
+    # been saved in production. Calling check_valid_file
+    # should now return "Duplicate title".
+    f = open(os.path.join(uploads, file_name_min), 'w')
+    f.write('why is file seek at the end???')
```
:frowning_face:

calebsander reviewed (changes requested):
Please also delete `donut/modules/uploads/.dummy`

koivyr committed:
Address PR feedback; no dup editing

koivyr committed:
Merge branch 'master' of https://github.com/ASCIT/donut into zmo/committee_sites

koivyr committed:
gitignore

koivyr committed:
Change gitignore fix some pr issues

koivyr committed:
Issues

koivyr committed:
Fix Styling + add case to delete pages from db + clean

koivyr committed:
Fixing never ending amount of problems rip

koivyr committed:
Small note about what file they can upload

koivyr committed:
Missed a few comments

koivyr committed:
Forgot about when files are removed

calebsander commented on .gitignore:
```diff
@@ -67,3 +67,8 @@ target/
 .DS_Store
+
+# Files related to uploading	
+donut/modules/uploads/uploaded_files/*
```
Don't actually need the `/*`; you can just ignore the whole directory

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+    If we are landing in the default page, automatically return True
+    """
+    # In seconds
+    TIMEOUT = 60 * 3
```
Should go at the top of the file

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, title)
+        res = cursor.fetchone()
```
Shouldn't need to `fetchone()` since you aren't fetching any values

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    curFile = read_file(os.path.join(path, name + '.md'))
+    return curFile
```
Nit, but Python convention is to use `snake_case` variable names

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+                        flask.current_app.config["UPLOAD_WEBPAGES"])
+    links = glob.glob(root + '/*')
+    results = []
+    for filenames in links:
```
`filenames` should just be `filename`

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+        filenames = filenames.replace(root + '/',
+                                      '').replace('.md', '').replace('_', ' ')
+        link = flask.url_for('uploads.display', url=filenames)
+        results.append((link, filenames))
```
Consider using a dict with clearly named elements instead of a tuple

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
+    for i in links:
+        name = i.replace(path + '/', '').replace('.md', '').replace("_", " ")
```
Consider abstracting this into a function since you do the same thing in `get_links()` too

Wingfril replied:
I think that'll be more trouble than it's worth since this one requires the os path to the file.

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+    """
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
```
Maybe make a `get_all_uploads()` function to do this globbing?

koivyr replied:
I still think it's more trouble than it's worth since each are a bit different

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,202 @@
+    Allows all numbers and characters. Allows ".", "_", "-"
+    """
+    title_res = re.match("^[0-9a-zA-Z.\/_\- ]*$", title)
+    return title_res != None and len(title) < 100
```
Consider using an `r` string for the regular expression. Also, can do the check for length first to avoid computing the regex match if not needed

koivyr replied:
re: check the length: I don't think it's necessary since it's only 100 characters

calebsander replied:
I just mean that it's faster to do the length check than the regex check, and you can stop as soon as either fails. So it makes more sense to check the length first because if it is too long, the regex match doesn't need to be computed.

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,120 @@
+    Returns the editor page where users can create and edit
+    existing pages
+    '''
+    input_text = "Hello world"
```
Don't think this is needed

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,120 @@
+from donut.modules.editor import blueprint, helpers
+
+
+@blueprint.route('/editor', methods=['GET', 'POST'])
```
Don't think this needs to support `POST`

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,120 @@
+        return flask.render_template(
+            'editor_page.html', input_text=input_text, title=title)
+    else:
+        return flask.abort(403)
```
Would be cleaner to switch the order of these:
```python
if helpers.is_locked(title, load_default) or not helpers.check_edit_page_permission():
    return flask.abort(403)

helpers.change_lock_status(title, True, load_default)
return flask.render_template('editor_page.html', input_text=input_text, title=title)
```

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,120 @@
+    markdown = flask.request.form['markdown']
+    title = flask.request.form['title']
+    markdown = markdown.replace('<', '&lt;')
+    markdown = markdown.replace('>', '&gt;')
```
Couple do both replacements in one statement

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+  old_title = $('#text_title').val();
+  $('#runBtn').click(preview);
+  $('#change_title').click(change_title);
+  for (var i = 1; i<7; i++) {
```
Might be clearer with `<= 6` instead

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+      converter = new showdown.Converter({strikethrough: true}),
+      html = converter.makeHtml(text);
+  target.html(html);
+  target_title.html(title);
```
Use `.text()` if you don't need to insert arbitrary HTML strings. This makes it less likely that there will be a cross-site scripting attack possible

koivyr replied:
We are inserting html

calebsander replied:
Into the title, though?

koivyr replied:
woops you right

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+function change_title(){
+  var title = $('#text_title').val();
+  var text = $('#source').val();
+  console.log(old_title);
```
Remove `console.log()`

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+        }
+      } else {
+        window.alert(data.error);
+      }
```
Don't think it really makes sense to `alert('Duplicate title')` if the user clicks "no" on the `confirm` dialog

koivyr replied:
The many brackets make it hard to read but the alert is for when data.error is not `Duplicate title` or ``

calebsander replied:
Ah, makes sense. Sorry I misread it

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+ });
+}, KEEP_PAGE_ALIVE_TIME);
+
+window.addEventListener("beforeunload", function (e) {
```
Don't need the `e` parameter

calebsander commented on donut/modules/editor/static/script.js:
```diff
@@ -0,0 +1,197 @@
+
+window.addEventListener("beforeunload", function (e) {
+  var title = $('#text_title').val();
+  var url = "{{ url_for('editor.close_page') }}";
```
`url` appears unused

calebsander commented on donut/modules/editor/static/style.css:
```diff
@@ -0,0 +1,18 @@
+#targetDiv {
+  border: 1px dashed #333333;
```
Can use `#333` as shorthand

calebsander commented on donut/modules/editor/templates/created_list.html:
```diff
@@ -0,0 +1,34 @@
+          {% for link, name in links %}
+            <tr>
+              <td><a href= "{{ link }}">{{name}}</a></td>
+              {% if permissions %}
```
Shouldn't have a second `<th>` if `permissions` is `False`

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,100 @@
+{% block page %} 
+<div class="container theme-showcase" role="main">
+  <div class="col-md-24">
+    <div class="jumbotron" style="text-align: center">
```
Would be nice to put this is a `<style>` tag instead

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,100 @@
+{% block scripts %}
+  {{super()}}
+  <script src="https://cdn.rawgit.com/showdownjs/showdown/1.0.1/dist/showdown.min.js"></script>
+  <script src={{ url_for('editor.static', filename='script.js', _external = True)}} type="text/javascript"></script>
```
Don't need `type="text/javascript"`

calebsander commented on donut/modules/editor/templates/editor_page.html:
```diff
@@ -0,0 +1,100 @@
+      <header>
+        <h1 id = "preview_title" class="text-center"></h1>
+      </header>
+      <div id="preview" style="text-align: left">
```
Isn't `text-align: left` the default?

koivyr replied:
Nope since I center the jumbotron (imo it looks better for that editor part to be centered). I'll put it in a css file though

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,91 @@
+    '''
+    Checks for allowed file extensions.
+    '''
+    return '.' in filename and \
```
Can just check whether `filename.rsplit('.', 1)` has length 2

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,91 @@
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def remove_link(filename):
```
Can this be combined with the similar function in `editor`? Similarly for `read_page()`

koivyr replied:
similar but not similar enough imo...the replace strings are different and probably a hassle to deal with

calebsander commented on donut/modules/uploads/helpers.py:
```diff
@@ -0,0 +1,91 @@
+    '''
+    file.seek(0, os.SEEK_END)
+    file_length = file.tell()
+    if file_length > 10 * 1024 * 1024:
```
Should make this a named constant

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,109 @@
+        permissions=helpers.check_upload_permission())
+
+
+@blueprint.route('/uploads/_send_page', methods=['GET'])
```
Don't think you need to restrict these URLs to `GET`

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,109 @@
+    '''
+    url = flask.request.args.get('url')
+    page = helpers.read_page(url.replace(' ', '_'))
+    return page
```
No need for `page` variable

koivyr replied:
This sends the content of the markdown file to the front end.

koivyr replied:
woops misunderstood you--sorry

calebsander commented on donut/modules/uploads/routes.py:
```diff
@@ -0,0 +1,109 @@
+    Handles the uploading of the file
+    '''
+    if 'file' not in flask.request.files:
+        return flask.abort(500)
```
`flask.abort()` raises an exception, so there is no need for `return`

calebsander commented on donut/modules/uploads/static/script.js:
```diff
@@ -0,0 +1,20 @@
+
+$(function() {
+  load();
+})
```
Can just do `$(load)`

calebsander commented on donut/modules/uploads/templates/uploaded_list.html:
```diff
@@ -0,0 +1,30 @@
+    </div>
+  </div>
+</div>
+{% endblock %}
```
This template looks very similar to the one for `editor`. Is it possible to merge them into one template?

koivyr replied:
hmmm by some sketchy ways (modifying `current_app.jinja_loader`) I can load this template in `editor`. However, it doesn't seem to be able to find `layout.html` from which it inherits from. 

So I'm going to move one of the routes to the other folder for now? There might be better ways that I can't seem to find.

calebsander replied:
Hmm, I think you might be able to symlink one of the templates to the other so there is only one copy

koivyr replied:
Yeee it worked

calebsander commented on donut/modules/uploads/templates/uploads.html:
```diff
@@ -0,0 +1,37 @@
+  <script src={{ url_for('uploads.static', filename='script_uploading.js', _external = True)}} type="text/javascript"></script>
+  <script type="text/javascript">
+    $SCRIPT_ROOT = {{ request.script_root|tojson|safe }};
+  </script>
```
Is this still necessary?

calebsander commented on donut/modules/uploads/uploaded_files/pages/Budgets.md:
```diff
@@ -0,0 +1,35 @@
+###YEARS AVAILABLE###
```
This file should be removed

calebsander commented on donut/modules/uploads/uploaded_files/pages/2018-2019_Minutes.md:
```diff
@@ -0,0 +1,9 @@
+[6 June 2018](6 June 2018)
```
Should remove this file

calebsander commented on donut/modules/uploads/uploaded_files/pages/Revcomm_Rulings.md:
```diff
@@ -0,0 +1,39 @@
+###**REVCOMM RULINGS**###
```
Should remove this file

calebsander reviewed (changes requested):
I think there are a few small things that could use some work. But `make test` is timing out for me. Is it working for you? The new files, especially the helpers files, are missing a fair amount of coverage.

calebsander commented on donut/__init__.py:
```diff
@@ -13,6 +13,7 @@
     from donut import default_config as config
 from donut import constants
+from donut.modules import account, auth, marketplace, core, courses, directory_search, editor, groups, rooms, uploads, voting
 from donut.modules import account, auth, marketplace, core, courses, directory_search, groups, rooms, voting
```
Should not be importing all these twice

calebsander commented:
Nvm, I hadn't pulled the most recent code

koivyr committed:
Stuck in fixing never ending pr feedbacks

koivyr committed:
Missed a file

koivyr committed:
Add symlink so we can reuse templates

koivyr committed:
Add symlink so we can reuse templates

koivyr committed:
Fixing unneeded checks + more testing

koivyr committed:
Fixing unneeded checks + more testing

koivyr committed:
Add smol new method to reduce redundancies

koivyr committed:
Add smol new method to reduce redundancies

koivyr committed:
Clean code

koivyr committed:
Clean code

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,227 @@
+
+    remove_file_from_db(old_filename)
+
+    if os.path.exists(old_path) and not os.path.exists(new_filename):
```
Should this be `new_path` instead of `new_filename`?

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -0,0 +1,227 @@
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    cur_file = read_file(os.path.join(path, name + '.md'))
+    return cur_file
```
Variable could be removed

calebsander commented on donut/modules/editor/routes.py:
```diff
@@ -0,0 +1,119 @@
+        input_text = helpers.read_markdown(inputt)
+        title = flask.request.args.get('title')
+        load_default = False
+    if inputt == None:
```
Could just use `else`

koivyr committed:
Incorrect checking path when renaming

koivyr committed:
:Merge branch 'zmo/committee_sites' of https://github.com/ASCIT/donut into zmo/committee_sites

calebsander commented on donut/modules/editor/helpers.py:
```diff
@@ -100,7 +100,6 @@ def rename_title(old_filename, new_filename):
     remove_file_from_db(old_filename)
 
     if os.path.exists(old_path) and not os.path.exists(new_path):
-        os.rename(old_path, new_path)
```
Why did this get removed? It's causing the tests to fail

koivyr replied:
whoops removed it on accident during merging. I'll add it back

calebsander commented:
I'll approve this if you can just revert the last commit, since it resulted in invalid Python. It would probably be good to write more extensive tests of the routes, but this is okay for now.

koivyr committed:
revert deletion
